var/spack/repos/builtin/packages/py-scikit-build-core/package.py

```diff
@@ -29,20 +55,21 @@ class PyScikitBuildCore(PythonPackage):
 
     # Dependencies
     depends_on("py-exceptiongroup", when="^python@:3.10", type=("build", "run"))
+    depends_on("py-importlib-metadata", type=("build", "run"), when="@0.3: ^python@:3.7")
     depends_on("py-importlib-resources@1.3:", when="^python@:3.8", type=("build", "run"))
     depends_on("py-packaging@20.9:", type=("build", "run"))
     depends_on("py-tomli@1.1:", when="^python@:3.10", type=("build", "run"))
     depends_on("py-typing-extensions@3.10:", when="^python@:3.7", type=("build", "run"))
     depends_on("cmake@3.15:", type=("build", "run"))
 
     # Optional dependencies
-    depends_on("py-pyproject-metadata@0.5:", when="+pyproject", type=("build", "run"))
     depends_on("py-pathspec@0.10.1:", when="+pyproject", type=("build", "run"))
+    depends_on("py-pyproject-metadata@0.5:", when="+pyproject", type=("build", "run"))
 
     # Test dependencies
     depends_on("py-build +virtualenv", type="test")
     depends_on("py-cattrs@22.2:", type="test")
-    depends_on("py-importlib-metadata", when="^python@:3.7", type="test")
+    depends_on("py-importlib-metadata", type=("build", "run"), when="@0.3: ^python@:3.7")
```

Comment:
This can revert to the test dependency it was before the change (since the build/run dependency is now reflected earlier).

Reply:
Done